resourcecustomizations/apps.kruise.io/v1alpha1/CloneSet/customizations.yaml

```diff
@@ -0,0 +1,320 @@
+            end
+            if statusItems[i].status ~= nil and statusItems[i].status.updateRevision ~= nil and statusItems[i].status.updateRevision ~= '' then
+              updateRevision = statusItems[i].status.updateRevision
+            end
+            if statusItems[i].status ~= nil and statusItems[i].status.currentRevision ~= nil and statusItems[i].status.currentRevision ~= '' then
+              currentRevision = statusItems[i].status.currentRevision
+            end
+          end
+          desiredObj.status.observedGeneration = generation
+          desiredObj.status.replicas = replicas
+          desiredObj.status.updatedReplicas = updatedReplicas
+          desiredObj.status.readyReplicas = readyReplicas
+          desiredObj.status.availableReplicas = availableReplicas
+          desiredObj.status.updatedReadyReplicas = updatedReadyReplicas
+          desiredObj.status.expectedUpdatedReplicas = expectedUpdatedReplicas
+          desiredObj.status.updateRevision = updateRevision
+          desiredObj.status.currentRevision = currentRevision
+          return desiredObj
+        end
+    statusReflection:
+      luaScript: >
+        function ReflectStatus (observedObj)
+          status = {}
+          if observedObj == nil or observedObj.status == nil then 
+            return status
+          end
+          status.replicas = observedObj.status.replicas
+          status.updatedReplicas = observedObj.status.updatedReplicas
+          status.readyReplicas = observedObj.status.readyReplicas
+          status.availableReplicas = observedObj.status.availableReplicas
+          status.updatedReadyReplicas = observedObj.status.updatedReadyReplicas
+          status.expectedUpdatedReplicas = observedObj.status.expectedUpdatedReplicas
+          status.updateRevision = observedObj.status.updateRevision
+          status.currentRevision = observedObj.status.currentRevision
+          return status
+        end
+    healthInterpretation:
+      luaScript: >
+        function InterpretHealth(observedObj)
+          return observedObj.status.readyReplicas == observedObj.spec.replicas
+          --[[
+          if observedObj.status.observedGeneration ~= observedObj.generation then
+            return false
+          end
+          if observedObj.spec.replicas ~= nil then
+            if observedObj.status.updatedReplicas < observedObj.spec.replicas then
+              return false
+            end
+          end
+          if observedObj.status.availableReplicas < observedObj.status.updatedReplicas then
+            return false
+          end
+          return true
+          --]]
+        end
+    dependencyInterpretation:
+      luaScript: >
+        function GetDependencies(desiredObj)
+          dependentConfigMaps = {}
+          dependentSecrets = {}
+          dependentSas = {}
+          dependentPVCs = {}
+          refs = {}
+          local idx = 1
+          if desiredObj.spec.template.spec.initContainers ~= nil then
```

Comment:
As you said, maybe we can introduce a new helper function `getDependencies From PodTemplate` to reduce duplicate code.

Reply:
Yes! If we add this function, It's also easy to complete `InterpretDependency` operation using lua, like `InterpretReplica` with `kube.accuratePodRequirements`.

Reply:
I agree with it. Would you like to send a separate PR to improve this? @yike21

Reply:
Yes, I'm willing to do it!

Reply:
As #3339 is merged,  we can start to refactor the codes about `interpretDependency`.

Reply:
OK, I will refactor the codes soon!